### ISSUE TITLE
"Slideshow" that slowly changes avatar image for a cluster

### ISSUE THREAD

**Issue by creativecoder** [Automattic/automatticians-map#8]
> It could work as a "slideshow" though, so the photos stay for a few seconds and then fade into the next one. If the animation is slow and subtle and the clusters have a random delay so they don't change all at once, it may look very nice (specially if you're looking the map from a low zoom level, because it would feel very alive 👌).

**Comment by creativecoder** [comment]
@evilluendas Here's a version with rotating cluster images. What do you think? Anything you'd adjust in terms of animation style or timing? https://automattic.github.io/automatticians-map/leaflet/

**Comment by evilluendas** [comment]
Looks great! 👍 

When I suggested this idea I was worried it could maybe make zooming in a cluster a bit confusing (imagine if the top avatar changes right when you are zooming in) but since they rotate randomly and reasonably fast, I think that would be a veeeeery unusual occurrence.

**Comment by creativecoder** [comment]
Know that the image rotation is currently set up to only show images that are available at the next zoom level for the cluster. So when you click on a cluster to zoom in, you'll always see the cluster's image somewhere on the map after the zoom completes. I think this is important to maintain a sense of consistency between zoom levels.

The tradeoff is that less images overall are shown as they rotate. For instance a cluster might say "4", but only rotate between 2 images. This is because at the next zoom level, that cluster breaks down into two features: a single marker and another cluster of 3.

**Comment by nb** [comment]
> Know that the image rotation is currently set up to only show images that are available at the next zoom level for the cluster.

Is the consistency the only reason we are doing that? Would it be easy to switch to rotating all images?

**Comment by creativecoder** [comment]
> Is the consistency the only reason we are doing that? Would it be easy to switch to rotating all images?

Yes, before implementing this way, I found it confusing when I clicked on an avatar and that avatar disappeared after zooming in. But if rotating through more images is a higher priority (to show more avatars overall), it would be pretty easy to switch. Those images are collected in a reduce function when the clusters are generated so we have full control on how to collect them.
